### DISCUSSION TITLE
search and replace without cursor jump

### DISCUSSION THREAD

**Discussion by alfieb123** [JetBrains/ideavim#520] [Q&A] [answered]
This seems like an issue that would have been discussed before, but for the life of me I cant find it. So, if this is an old question, I apologize.

When doing a search and replace:
:s/foo/bar/g (and its like)
my cursor jumps all over the editor to find the replaceable term. This is not only visually jarring, it means i have to remember to take note of whatever line movement I want to do, as when the cursor jumps line I have to guess the movement I wanted.

How can i stop this behaviour? 
Thank you kindly...

**Comment by AlexPl292** [comment, marked as answer]
Hello! I'm not sure that I understand your question. Do you mean that the caret jumps to the changed item after the replace? In original vim it works the same
![2022-06-24 18 36 21](https://user-images.githubusercontent.com/4203721/175569348-24ea26dd-bed7-495b-99eb-5699208b4e3d.gif)

**Reply by alfieb123** [reply]
Hi Alex. 
Thanks so much for your reply.
No that's not it...Sorry I wasn't clear.

I mean that as I start to type whatever term I will substitute, the caret jumps all over the file to progressively match that search term.
So if I were to start typing:
:s/foo/bar/g
as soon as I start typing 'foo' my caret jumps to match all 'f' then all 'fo' then all 'foo' etc..
If I'm not being clear enough Ill try and make a gif like you have done :)

**Reply by AlexPl292** [reply]
Oh, I suppose I understand what you mean. This feature is called `incsearch`, but it's disabled by default. Please check if you have `set incsearch` in your `.ideavimrc` and try to remove it.
If I understood your incorrect, probably you mean that after this jumping you press esc and you appear not on the place where you start the search? If that is the case, this is a but, IdeaVim should return the caret on the initial place.

![2022-06-24 21 03 12](https://user-images.githubusercontent.com/4203721/175617830-ab45cc30-4d9b-4716-b673-f43474acf6c6.gif)

**Reply by alfieb123** [reply]
Oh lordy lordy...
That's exactly what it was. Thanks so much for your help. 
Whoops.
